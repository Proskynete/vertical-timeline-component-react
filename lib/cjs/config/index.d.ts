import { DefaultTimelineProps } from '../interfaces';
export declare const defaultValues: DefaultTimelineProps;
export declare const mapText: {
    de: {
        from: string;
        to: string;
    };
    en: {
        from: string;
        to: string;
    };
    es: {
        from: string;
        to: string;
    };
    tr: {
        from: string;
        to: string;
    };
    zh: {
        from: string;
        to: string;
    };
};
