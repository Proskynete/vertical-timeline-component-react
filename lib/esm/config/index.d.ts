export interface Theme {
    yearColor: string;
    lineColor: string;
    dotColor: string;
    borderDotColor: string;
    titleColor: string;
    subtitleColor: string;
    textColor: string;
}
export declare type DateFormat = 'only-number' | 'short' | 'with-weekday' | 'full';
export interface DefaultValues {
    theme?: Theme;
    lang: string;
    dateFormat: DateFormat;
}
export declare const defaultValues: DefaultValues;
export declare const mapText: {
    en: {
        from: string;
        to: string;
    };
    es: {
        from: string;
        to: string;
    };
    de: {
        from: string;
        to: string;
    };
    tr: {
        from: string;
        to: string;
    };
    zh: {
        from: string;
        to: string;
    };
};
