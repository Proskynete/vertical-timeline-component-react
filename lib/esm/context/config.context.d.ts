import React, { PropsWithChildren } from 'react';
import { DefaultTimelineProps } from '../interfaces';
export interface ConfigProviderProps {
    config: DefaultTimelineProps;
}
declare const ConfigContext: React.Context<ConfigProviderProps>;
export declare const ConfigProvider: ({ children, config }: PropsWithChildren<ConfigProviderProps>) => JSX.Element;
export declare const ConfigCustomer: React.Consumer<ConfigProviderProps>;
export { ConfigContext };
