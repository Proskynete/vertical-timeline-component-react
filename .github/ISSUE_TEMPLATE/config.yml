contact_links:
  - name: Ask the community
    url: https://github.com/Proskynete/vertical-timeline-component-react/discussions/new
    about: Ask and discuss questions with other members of the community.
