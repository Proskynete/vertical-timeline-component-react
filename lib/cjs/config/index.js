"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.mapText = exports.defaultValues = void 0;
exports.defaultValues = {
    collapse: false,
    dateFormat: 'only-number',
    descriptionSymbol: '•',
    lang: 'en',
    theme: {
        borderDotColor: '#ffffff',
        descriptionColor: '#cccccc',
        dotColor: '#c5c5c5',
        eventColor: '#cccccc',
        lineColor: '#c5c5c5',
        subtitleColor: '#c5c5c5',
        titleColor: '#cccccc',
        yearColor: '#888888',
    },
    withoutDay: false,
};
exports.mapText = {
    de: { from: 'Von', to: 'Bis' },
    en: { from: 'From', to: 'To' },
    es: { from: 'Desde', to: 'Hasta' },
    tr: { from: 'Başlangıç', to: 'Bitiş' },
    zh: { from: '从', to: '到' },
};
