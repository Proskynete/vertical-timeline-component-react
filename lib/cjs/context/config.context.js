"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConfigContext = exports.ConfigCustomer = exports.ConfigProvider = void 0;
var react_1 = __importStar(require("react"));
var config_1 = require("../config");
var ConfigContext = (0, react_1.createContext)({
    config: config_1.defaultValues,
});
exports.ConfigContext = ConfigContext;
var ConfigProvider = function (_a) {
    var children = _a.children, config = _a.config;
    return react_1.default.createElement(ConfigContext.Provider, { value: { config: config } }, children);
};
exports.ConfigProvider = ConfigProvider;
exports.ConfigCustomer = ConfigContext.Consumer;
