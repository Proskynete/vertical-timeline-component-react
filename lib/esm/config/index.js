export var defaultValues = {
    collapse: false,
    dateFormat: 'only-number',
    descriptionSymbol: '•',
    lang: 'en',
    theme: {
        borderDotColor: '#ffffff',
        descriptionColor: '#cccccc',
        dotColor: '#c5c5c5',
        eventColor: '#cccccc',
        lineColor: '#c5c5c5',
        subtitleColor: '#c5c5c5',
        titleColor: '#cccccc',
        yearColor: '#888888',
    },
    withoutDay: false,
};
export var mapText = {
    de: { from: 'Von', to: 'Bis' },
    en: { from: 'From', to: 'To' },
    es: { from: 'Desde', to: 'Hasta' },
    tr: { from: 'Başlangıç', to: 'Bitiş' },
    zh: { from: '从', to: '到' },
};
