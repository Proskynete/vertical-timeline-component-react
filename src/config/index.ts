export interface Theme {
	yearColor: string;
	lineColor: string;
	dotColor: string;
	borderDotColor: string;
	titleColor: string;
	subtitleColor: string;
	textColor: string;
}

export type DateFormat = 'L' | 'l' | 'll';

export interface DefaultValues {
	theme?: Theme;
	lang: string;
	dateFormat: DateFormat;
}

export const defaultValues: DefaultValues = {
	theme: {
		yearColor: '#888888',
		lineColor: '#c5c5c5',
		dotColor: '#c5c5c5',
		borderDotColor: '#ffffff',
		titleColor: '#cccccc',
		subtitleColor: '#888888',
		textColor: '#cccccc',
	},
	lang: 'en',
	dateFormat: 'L',
};

export const monthArray = {
	es: [
		'Ene',
		'Feb',
		'Mar',
		'Abr',
		'May',
		'Jun',
		'Jul',
		'Ago',
		'Sep',
		'Oct',
		'Nov',
		'Dic',
	],
	en: [
		'Jan',
		'Feb',
		'Mar',
		'Apr',
		'May',
		'Jun',
		'Jul',
		'Aug',
		'Sep',
		'Oct',
		'Nov',
		'Dec',
	],
	de: [
		'Jan.',
		'Feb.',
		'März',
		'Apr.',
		'Mai.',
		'Juni',
		'Juli',
		'Aug.',
		'Sep.',
		'Okt.',
		'Nov.',
		'Dez.',
	],
	tr: [
		'Ocak',
		'Şubat',
		'Mart',
		'Nisan',
		'Mayıs',
		'Haziran',
		'Temmuz',
		'Ağustos',
		'Eylül',
		'Ekim',
		'Kasım',
		'Aralık',
	],
	zh_CN: [
		'一月',
		'二月',
		'三月',
		'四月',
		'五月',
		'六月',
		'七月',
		'八月',
		'九月',
		'十月',
		'十一月',
		'十二月',
	],
};

export const mapText = {
	en: {
		from: 'From',
		to: 'To',
	},
	es: {
		from: 'Desde',
		to: 'Hasta',
	},
	de: {
		from: 'Von',
		to: 'Bis',
	},
	tr: {
		from: 'Başlangıç',
		to: 'Bitiş',
	},
	zh_CN: {
		from: '从',
		to: '到',
	},
};
