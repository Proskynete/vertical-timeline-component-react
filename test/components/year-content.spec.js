import React from 'react';
import Enzyme, { render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

import YearContent from '../../src/components/year-content';
import { ConfigContext } from '../../src/context/config.context';

Enzyme.configure({ adapter: new Adapter() });

describe('<YearContent />', () => {
	it('should create Snapshot', () => {
		const component = renderer.create(
			<ConfigContext.Provider value={{ lang: 'en', dateFormat: 'L' }}>
				<YearContent startDate='2020/08/21' />
			</ConfigContext.Provider>,
		);

		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('should render component with startDate props', () => {
		const wrapper = render(
			<ConfigContext.Provider value={{ lang: 'en', dateFormat: 'L' }}>
				<YearContent startDate='2020/08/21' />
			</ConfigContext.Provider>,
		);

		expect(wrapper.find('span').text()).toEqual('From');
		expect(wrapper.find('time').text()).toEqual('08/21/2020');
	});

	it('should render component with startDate and currentYear props', () => {
		const wrapper = render(
			<ConfigContext.Provider value={{ lang: 'en', dateFormat: 'L' }}>
				<YearContent startDate='2019/08/21' currentYear />
			</ConfigContext.Provider>,
		);

		expect(wrapper.find('span')).toHaveLength(1);
		expect(wrapper.find('span').text()).toEqual('From');

		expect(wrapper.find('time')).toHaveLength(2);
		expect(wrapper.find('time').text().includes('2021')).toBeTruthy();
		expect(wrapper.find('time').text().includes('08/21/2019')).toBeTruthy();
	});

	it('should render component with startDate and endDate props', () => {
		const wrapper = render(
			<ConfigContext.Provider value={{ lang: 'en', dateFormat: 'L' }}>
				<YearContent startDate='2019/08/21' endDate='2020/08/21' />
			</ConfigContext.Provider>,
		);

		expect(wrapper.find('span')).toHaveLength(2);
		expect(wrapper.find('span').text().includes('From')).toBeTruthy();
		expect(wrapper.find('span').text().includes('To')).toBeTruthy();

		expect(wrapper.find('time')).toHaveLength(2);
		expect(wrapper.find('time').text().includes('08/21/2020')).toBeTruthy();
		expect(wrapper.find('time').text().includes('08/21/2019')).toBeTruthy();
	});
});
