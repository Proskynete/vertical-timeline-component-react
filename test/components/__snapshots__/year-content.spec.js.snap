// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<YearContent /> should create Snapshot 1`] = `
<p
  className="sc-eCssSg eJxYfY"
  format="L"
  lang="en"
>
  <span
    className="sc-jSgupP hSlCmR"
  >
    From
  </span>
  <time
    dateTime="2020/08/21"
  >
    08/21/2020
  </time>
</p>
`;
